packages/astro/src/default/components/OpenInStackblitzLink.astro

```diff
@@ -0,0 +1,93 @@
+<button
+  transition:persist
+  title="Open in StackBlitz"
+  data-id="open-in-stackblitz"
+  class="flex items-center font-size-3.5 text-tk-elements-topBar-iconButton-iconColor hover:text-tk-elements-topBar-iconButton-iconColorHover transition-theme bg-tk-elements-topBar-iconButton-backgroundColor hover:bg-tk-elements-topBar-iconButton-backgroundColorHover p-1 rounded-md"
+>
+  <svg viewBox="0 0 28 28" aria-hidden="true" height="24" width="24">
+    <path fill="currentColor" d="M12.747 16.273h-7.46L18.925 1.5l-3.671 10.227h7.46L9.075 26.5l3.671-10.227z"></path>
+  </svg>
+</button>
+
+<script>
+  import StackBlitzSDK from '@stackblitz/sdk';
+  import { tutorialStore } from '../components/webcontainer.js';
+
+  // initialize handlers on each page load as it's possible some pages disable openInStackBlitzLink
+  document.addEventListener('astro:page-load', onInit);
+
+  function onInit() {
+    const buttons = document.querySelectorAll('[data-id="open-in-stackblitz"]');
+    buttons.forEach((button) => button.addEventListener('click', onClick));
+  }
+
+  function onClick() {
+    const lesson = tutorialStore.lesson;
+
+    if (!lesson) {
+      throw new Error('Missing lesson');
+    }
+
+    const files: Record<string, string> = {};
+
+    // first add template files
+    for (const [filePath, value] of Object.entries(tutorialStore.template || {})) {
+      files[removeLeadingSlash(filePath)] = value.toString();
+    }
+
+    // next overwrite with files from editor
+    for (const { filePath, value } of Object.values(tutorialStore.documents.get())) {
+      files[removeLeadingSlash(filePath)] = value.toString();
```

Comment:
Images mostly but any non-text data that users could endup having in WebContainer. I think you should skip elements for now. I don't know if we have a way so that they're processed correctly on StackBlitz atm.